File: Frontend/src/types/job.ts
export type ApplicationStatus =
  | 'saved'
  | 'applied'
  | 'interviewing'
  | 'offered'
  | 'accepted'
  | 'rejected';

export interface Job {
  id: number;
  title: string;
  company: string;
  company_logo?: string;
  location: string;
  salary?: string;
  description: string;
  requirements?: string[];
  tags?: string[];
  remote?: boolean;
  source?: string;
  posted_at?: string;
  saved?: boolean;
}

export interface JobsResponse {
  jobs: Job[];
  total: number;
}

export interface JobStats {
  total: number;
  remote: number;
  internships: number;
  saved: number;
  applied: number;
}

export interface Application {
  id: number;
  job_id: number;
  job_title: string;
  company: string;
  status: ApplicationStatus;
  ats_score?: number;
  applied_at?: string;
  interview_date?: string | null;
  notes?: string;
}

export interface ResumeExperience {
  company?: string;
  role?: string;
  start_date?: string;
  end_date?: string;
  description?: string;
}

export interface ResumeEducation {
  school?: string;
  degree?: string;
  field?: string;
  start_date?: string;
  end_date?: string;
}

export interface ResumeProject {
  name?: string;
  description?: string;
  url?: string;
}

export interface ResumeData {
  full_name: string;
  email: string;
  phone: string;
  summary: string;
  skills: string[];
  experience: ResumeExperience[];
  education: ResumeEducation[];
  projects: ResumeProject[];
  portfolio_url: string;
  github_url: string;
  linkedin_url: string;
}

export interface ATSResult {
  score: number;
  strengths: string[];
  weaknesses: string[];
  suggestions: string[];
  overall_feedback: string;
}

export interface AssistantMessage {
  role: 'user' | 'assistant' | 'system';
  content: string;
}

export interface AssistantResponse {
  response: string;
  suggestions?: string[];
  extracted_fields?: Record<string, unknown>;
}

File: Backend/app/schemas/job.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class JobResponse(BaseModel):
    id: int
    title: str
    company: str
    company_logo: str
    location: str
    salary: str
    description: str
    requirements: List[str] = []
    tags: List[str] = []
    remote: bool
    source: str
    posted_at: datetime
    
    class Config:
        from_attributes = True


class JobListResponse(BaseModel):
    jobs: List[JobResponse]
    total: int
    page: int
    limit: int


class JobSearchRequest(BaseModel):
    query: Optional[str] = None
    role: Optional[str] = None
    location: Optional[str] = None
    remote: Optional[bool] = None
    company: Optional[str] = None
    tags: List[str] = []
    page: int = 1
    limit: int = 20


class ResumeData(BaseModel):
    full_name: str = ""
    email: str = ""
    phone: str = ""
    summary: str = ""
    skills: List[str] = []
    experience: List[dict] = []
    education: List[dict] = []
    projects: List[dict] = []
    certifications: List[dict] = []
    portfolio_url: str = ""
    github_url: str = ""
    linkedin_url: str = ""


class ATSReviewResponse(BaseModel):
    score: int
    strengths: List[str] = []
    weaknesses: List[str] = []
    suggestions: List[str] = []
    keyword_match: int = 0
    format_score: int = 0
    content_score: int = 0
    overall_feedback: str = ""


class ApplicationResponse(BaseModel):
    id: int
    job_id: int
    job_title: str = ""
    company: str = ""
    status: str
    ats_score: int = 0
    applied_at: Optional[datetime] = None
    interview_date: Optional[datetime] = None
    notes: str = ""
    created_at: datetime
    
    class Config:
        from_attributes = True


class ApplicationStats(BaseModel):
    total_applications: int
    applied: int
    interviewing: int
    offered: int
    accepted: int
    rejected: int
    response_rate: float


class CareerAdviceRequest(BaseModel):
    question: str
    context: Optional[str] = None